### PR TITLE
fix(storage): fix object store config

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We mistakenly used the default value for the object store config.

The current PR fixes this bug so that the object store can read the storage config for initialization.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by zwang28** [comment]
But `S3ObjectStoreConfig` indeed only applies to S3, not OpenDAL.

File: src/ctl/src/common/hummock_service.rs [hunk 1]
```diff
@@ -66,16 +67,15 @@ impl HummockServiceOpts {
...
-For `./risedev d` use cases, please do the following.
-* start the cluster with shared storage:
-  - consider adding `use: minio` in the risedev config,
-  - or directly use `./risedev d for-ctl` to start the cluster.
-* use `./risedev ctl` to use risectl.
-
-For `./risedev apply-compose-deploy` users,
-* `RW_HUMMOCK_URL` will be printed out when deploying. Please copy the bash exports to your console.
-";
+                    For `./risedev d` use cases, please do the following.
```
Comment (zwang28) [inline src/ctl/src/common/hummock_service.rs:70]: Is this formatting by intention?

**Review by zwang28 — APPROVED** [review]
rest LGTM. Thanks!

**Review by wcy-fdu — APPROVED** [review]
LGTM

**Review by MrCroxx — APPROVED** [review]

**Pull request merged**